File: app/src/main/java/com/jsqix/dongqing/gank/fragment/douban/DoubanFragment.java
package com.jsqix.dongqing.gank.fragment.douban;

import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;

import com.jsqix.dongqing.gank.bean.HomeData;
import com.jsqix.dongqing.gank.fragment.BaseHomeActivity;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by dongqing on 2017/1/12.
 */

public class DoubanFragment extends BaseHomeActivity {

    @Override
    protected HomeData setUpViewPage() {
        mTabLayout.setTabMode(TabLayout.MODE_FIXED);
        List<String> titles = new ArrayList<>();
        titles.add("正在热映");
        titles.add("即将上线");
        List<Fragment> fragments = new ArrayList<>();
        for (int i = 0; i < titles.size(); i++) {
            Fragment fragment = new DoubanFilmFragment();
            Bundle bundle = new Bundle();
            bundle.putString("type", titles.get(i));
            fragment.setArguments(bundle);
            fragments.add(fragment);
        }
        return new HomeData(titles, fragments);
    }
}

File: app/src/main/java/com/dongqing/gank/kotlin/ui/fragment/douban/DoubanFragment.kt
package com.dongqing.gank.kotlin.ui.fragment.douban

import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v4.app.Fragment
import com.dongqing.gank.kotlin.bean.HomeData
import com.dongqing.gank.kotlin.ui.fragment.BaseHomeFragment
import kotlinx.android.synthetic.main.fragment_home.*
import java.util.*

class DoubanFragment : BaseHomeFragment() {
    override fun setUpViewPage(): HomeData {
        tabs.tabMode = TabLayout.MODE_FIXED
        val titles = ArrayList<String>()
        titles.add("正在热映")
        titles.add("即将上线")
        val fragments = ArrayList<Fragment>()
        for (i in titles.indices) {
            val fragment = DoubanFilmFragment()
            val bundle = Bundle()
            bundle.putString("type", titles[i])
            fragment.arguments = bundle
            fragments.add(fragment)
        }
        return HomeData(titles, fragments)
    }
}